### PR TITLE
Refactor/refactoring of statistics commands

### PR DESCRIPTION
**Pull request by MatiF100**
Fixed problems within all stats commands. `statsall` and `updateStatsAll` should now take appropriate amount of RAM

### CODE REVIEW

File: SKNIBot/Core/Bot.cs [hunk 1]
```diff
@@ -18,10 +18,13 @@
 using Microsoft.Extensions.Logging;
 using SKNIBot.Core.Services.RolesService;
 using DSharpPlus.Exceptions;
+using SKNIBot.Core.Database.Models.DynamicDB;
```
Comment (AzuxDario) [inline SKNIBot/Core/Bot.cs:21]: Is it necessary?

File: SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs [hunk 2]
```diff
@@ -1,181 +1,381 @@
...
 using System.Linq;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using DSharpPlus;
 using DSharpPlus.CommandsNext;
 using DSharpPlus.CommandsNext.Attributes;
 using DSharpPlus.Entities;
 using DSharpPlus.Exceptions;
+using SKNIBot.Core.Database.Models.DynamicDB;
```
Comment (AzuxDario) [inline SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs:12]: Is it necessary? Commands shouldn't use DB directly.

File: SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs [hunk 3]
```diff
@@ -1,181 +1,381 @@
...
+            _cacheReloadTimer = new Timer(CacheReloadCallback, null, 40000, Timeout.Infinite);
+        }
+
+        private async void CacheReloadCallback(object state)
+        {
+            _cacheReloadTimer.Change(CacheReloadInterval, Timeout.Infinite);
+            Console.Out.WriteLine("Starting full stats cache reload... This will take a lot of time...");
+            var servers = _statsHookUser.GetServerList();
+            servers.UnionWith(_statsHookDate.GetServerList());
+            foreach (var server in servers)
```
Comment (AzuxDario) [inline SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs:45]: What if database has ServerId where bot is no longer present? We don't delete old servers from database. Won't this cause an exception?
And maybe we should reload cache only for server where command was executed.

File: SKNIBot/Core/Database/Models/DynamicDB/DateMessageStats.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        }
+        public DateMessageStat(int cnt, ulong channel, string date)
+         {
+             count = cnt;
+             ChannelID = channel;
+             Date = date;
+         }
+        public int ID { get; set; }
+        public int ServerID { get; set; }
+        public ulong ChannelID { get; set; }
```
Comment (AzuxDario) [inline SKNIBot/Core/Database/Models/DynamicDB/DateMessageStats.cs:19]: In SQLite Integer can use 8 bytes, but it's signed integer. Are you sure putting there ulong won't overflow it causing issues?
https://www.sqlite.org/datatype3.html
Previously we used string to keep ulong Discord ids.

File: SKNIBot/Core/Database/Models/DynamicDB/UserMessageStats.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        }
+        public UserMessageStat(int cnt, ulong channel, ulong user)
+         {
+             count = cnt;
+             ChannelID = channel;
+             User = user;
+         }
+        public int ID { get; set; }
+        public int ServerID { get; set; }
+        public ulong ChannelID { get; set; }
```
Comment (AzuxDario) [inline SKNIBot/Core/Database/Models/DynamicDB/UserMessageStats.cs:19]: In SQLite Integer can use 8 bytes, but it's signed integer. Are you sure putting there ulong won't overflow it causing issues?
https://www.sqlite.org/datatype3.html
Previously we used string to keep ulong Discord ids.

File: SKNIBot/Core/Database/Models/DynamicDB/UserMessageStats.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        public ulong User { get; set; }
```
Comment (AzuxDario) [inline SKNIBot/Core/Database/Models/DynamicDB/UserMessageStats.cs:20]: Same as ChannelID. And if it's id this should be called UserID,

**Review by AzuxDario — CHANGES_REQUESTED** [review]
Please refer to comments. And check if DynamicDatabase is really empty. It's template but growed about 120%.

File: SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs (hunk 3 above)
Comment (MatiF100) [reply SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs:43]: Fixed by applying exception handling with cache invalidation

**Review by AzuxDario — APPROVED** [review]
Ok

**Pull request merged**